src/main/java/spoon/support/compiler/jdt/CompilationUnitComparator.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.support.compiler.jdt;
+
+import org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration;
+import spoon.Launcher;
+
+import java.util.Comparator;
+import java.util.Random;
+
+public class CompilationUnitComparator implements Comparator<CompilationUnitDeclaration> {
+
+	@Override
+	public int compare(CompilationUnitDeclaration o1, CompilationUnitDeclaration o2) {
+		int seed = 0;
+		if (System.getenv("SPOON_SEED_CU_COMPARATOR") != null) {
```

Comment:
> copied from javadoc of System#getenv
> If a security manager exists, its checkPermission method is called with a RuntimePermission("getenv."+name) permission. This may result in a SecurityException being thrown. 

It would be cleaner to try catch security exception here.